### PR TITLE
added solrcloud functionality for better load balancing

### PR DESCRIPTION
**Pull request by harspras**
With reference to #156 I have added option to enable SolrCloud features in Banana. The dashboard settings will allow users to add the path of a [Zookeeper rest server](https://github.com/apache/zookeeper/tree/trunk/src/contrib/rest) which will be queried to get a list of available Solr nodes. 
##### The UI will look as follows:

(Updated) 
screen shot 2015-03-25 at 10 18 46

screen shot 2015-03-25 at 10 18 37
### [UPDATE]

For the time being, to use Zookeeper rest server, you will have to enable proxy otherwise you might face Cross Origin Resource Sharing error. I will soon add a patch for enabling Cross Origin in Zookeeper rest server under resources directory.

### CODE REVIEW

File: src/app/partials/dasheditor.html [hunk 1]
```diff
@@ -144,9 +144,30 @@
   <!-- Solr tab -->
   <div ng-show="editor.index == 3">
     <div class="row-fluid">
-      <h4>Solr Settings</h4>
+      <h4>Server Settings</h4>
     </div>
     <div class="row-fluid">
+      <h6>Choose the server</h6>
```
Comment (fxprunayre) [inline src/app/partials/dasheditor.html:150]: I would have put the solr server option on the left, as it will probably be the default (and most used) ?

Comment (harspras) [inline src/app/partials/dasheditor.html:150]: Yeah, I agree too. I will change that.

**Comment by aadel** [comment]
@harspras thanks for contributing this cool feature! However, please note that new features should go to dev-2.0 branch.

**Comment by harspras** [comment]
Ohh. Is that so. The Contributing.MD only mentioned develop branch. Do you want me to move the changes to dev-2.0? I can do that tomorrow.

**Comment by aadel** [comment]
Yes, that would be great.

**Comment by harspras** [comment]
@aadel Can you help me with regard to moving changes to dev-2.0 branch. The dev-2.0 uses bower_components to find dependencies, but as far as I know, bower is part of SILK stack. Is there any way to disable bower use so that I can test if everything is working fine on dev-2.0?

**Comment by aadel** [comment]
@harspras it's easier to install Bower than disabling it. Please check README.md of dev-2.0 for more details on getting it running.

**Comment by harspras** [comment]
Yeah, I commented too soon. I have made my changes. Will submit a new pull request.

**Comment by harspras** [comment]
Will continue this contribution in Pull Request #160 . Closing this for now.

**Comment by dougmcclure** [comment]
Our default is to run in SolrCloud with Zookeeper. Could you share the
benefits of running with these new changes you've implemented compared to
just running with Solr server?  What would be different? Performance,
response time, certain queries work better?

Tks!

**Comment by harspras** [comment]
The reason for adding zookeeper functionality too along with solr server node are many.
1.   To prevent a heavy dashboard (maybe with many faceted queries) affect other user’s queries/dashboards.
2.   To have high availability with minimal user interference. If the solr node the banana dashboard is pointing to is down, then the banana dashboard is not available. The zookeeper on the other hand can be used to select a replica or another node automatically in such scenarios.
3.   If there are too many banana dashboards pointed to the same server, then there is a chance that only one server cannot handle all the incoming queries. In that scenario, either you can change the other dashboards to point to different servers, or you can choose the zookeeper so that it changes the querying solr server dynamically.

There might be other use cases too.

**Pull request closed without merging**